src/main/java/com/mymemor/mymemor/controller/SearchController.java

```diff
@@ -1,16 +1,62 @@
 package com.mymemor.mymemor.controller;
 
-import org.springframework.web.bind.annotation.PostMapping;
+import com.mymemor.mymemor.exceptions.EmptyQuery;
+import com.mymemor.mymemor.model.User;
+import com.mymemor.mymemor.repository.AccountRepository;
+import com.mymemor.mymemor.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
 @RestController
 @RequestMapping("/api")
 public class SearchController {
 
-    @PostMapping("/search")
-    public String getSearchResultViaAjax() {
-        return "";
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private AccountRepository accountRepository;
+    @PersistenceContext
+    private EntityManager em;
+
+    private List<User> getUserById(List o){
+        List<User> list_user = new ArrayList<>();
+        for(Object id: o){
+            User user = userRepository.findById(((BigInteger) id).longValue()).orElseThrow();
+            list_user.add(user);
+        }
+        return list_user;
+    }
+
+    private List getResultBasedOnUsername(EntityManager em, String query){
```

Comment:
Change name of function.
```suggestion
    private List getUserIdsByUsername(EntityManager em, String query){
```